src/Server/WebSocketServer.php

```diff
@@ -162,10 +162,47 @@ private function onNewConnection(ConnectionInterface $socketStream)
             return $this->getMessageHandler($uri, $connection);
         }, $this->loop, $this->messageProcessor);
 
+        $socketStream->on('end', function () use($connection) {
+            $this->onDisconnect($connection);
+        });
+
         $connection->setLogger($this->getLogger());
+        $connection->getLogger()->info(sprintf('Ip "%s" establish connection', $connection->getIp()));
         $this->connections[] = $connection;
     }
 
+    /**
+     *
+     * @param Connection $connection
+     */
+    private function onDisconnect(Connection $connection)
+    {
+        $this->removeConnection($connection);
+        $connection->getLogger()->info(sprintf('Ip "%s" left connection', $connection->getIp()));
+    }
+
+    /**
+     * Remove a Connection instance by his object id
+     * @param Connection        $connection
+     * @param bool              $strict Define if the process should stop if unfoundable
+     * @throws RuntimeException This method throw an exception if the $connection instance object isn't findable in websocket server's connections
+     */
+    private function removeConnection(Connection $connection, bool $strict = true)
+    {
+        $connectionId = spl_object_hash($connection);
+        foreach ($this->connections as $index => $connectionIt) {
```

Comment:
Typo in `$connectionIt`.
In order to distinct the variable of the line above you can also name it `$connectionToRemove`.

Reply:
Actually, it isn't a typo, but a bad naming